symplyphysics/laws/vector/cosine_of_angle_between_vectors.py

```diff
@@ -0,0 +1,57 @@
+from sympy import Eq, Symbol as SymSymbol, cos, solve
```

Comment:
I would argue that the planar projection law is also inherently mathematical, but we still have it. The main reason for this PR is the proof for law about the linear displacement during rotation, see [this](https://github.com/blackyblack/symplyphysics/pull/317#discussion_r1470060128). So I figured I might as well move it to a separate law. I do believe this would probably not be needed if sympy were able to reason about such vector equalities in a symbolic way.

Reply:
Yes, planar projection is also pure math and should not belong here. Sometimes it may have a physical meaning though, for example when light casts a shadow on the ground.